warehouse/src/main/java/org/apache/hertzbeat/warehouse/service/impl/MetricsDataServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hertzbeat.warehouse.service.impl;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.hertzbeat.common.constants.CommonConstants;
+import org.apache.hertzbeat.common.entity.dto.Field;
+import org.apache.hertzbeat.common.entity.dto.MetricsData;
+import org.apache.hertzbeat.common.entity.dto.MetricsHistoryData;
+import org.apache.hertzbeat.common.entity.dto.Value;
+import org.apache.hertzbeat.common.entity.dto.ValueRow;
+import org.apache.hertzbeat.common.entity.message.CollectRep;
+import org.apache.hertzbeat.common.support.exception.CommonException;
+import org.apache.hertzbeat.warehouse.service.MetricsDataService;
+import org.apache.hertzbeat.warehouse.store.history.HistoryDataReader;
+import org.apache.hertzbeat.warehouse.store.realtime.RealTimeDataReader;
+import org.springframework.stereotype.Service;
+
+/**
+ *
+ */
+@Service
+public class MetricsDataServiceImpl implements MetricsDataService {
+
+    private final RealTimeDataReader realTimeDataReader;
+
+    private final Optional<HistoryDataReader> historyDataReader;
+
+    public MetricsDataServiceImpl(RealTimeDataReader realTimeDataReader, Optional<HistoryDataReader> historyDataReader) {
+        this.realTimeDataReader = realTimeDataReader;
+        this.historyDataReader = historyDataReader;
+    }
+
+    @Override
+    public Boolean getWarehouseStorageServerStatus() {
+        return historyDataReader.isPresent() && historyDataReader.get().isServerAvailable();
+    }
+
+    @Override
+    public MetricsData getMetricsData(Long monitorId, String metrics) {
+        boolean available = realTimeDataReader.isServerAvailable();
+        if (!available) {
+            throw new CommonException("real time store not available");
```

Comment:
Looks like this `CommonException` is not catched by `MetricsDataController`.

Reply:
`CommonException` is caught by the `GlobalExceptionHandler` in `Manager`, making sure it is the same as before, and throwing an exception forward when it is not available